challenges/Solution 2/TestProject/src/restApi/RestApiCall.java

```diff
@@ -0,0 +1,94 @@
+package restApi;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.sun.jersey.multipart.FormDataParam;
+
+@Path("/rest")
+public class RestApiCall {
+
+	private static List<Integer> numberListInt;
+	private static final int expectedListSize = 500;
+
+	@POST
+	@Consumes(MediaType.MULTIPART_FORM_DATA)
+	@Path("/data")
+	public String validateNumbersPost(@FormDataParam("file") InputStream fileInputStream) {
+		
+		JSONObject numberListJsonObj = null;
+		JSONArray jsonIntArray = null;
+		String jsonString = null;
+		String actualJsonString = null;
+		List<Integer> numberList = new ArrayList<>();
+		
+		try {
+			ByteArrayOutputStream result = new ByteArrayOutputStream();
+			byte[] buffer = new byte[1024];
+			int length;
+			while ((length = fileInputStream.read(buffer)) != -1) {
+				result.write(buffer, 0, length);

```

Comment:
Could you explain what you're doing with the `fileInputStream`? Are you storing the number data in a file on disk for permanent storage?